pyproject.toml

```diff
@@ -84,9 +84,10 @@ test = [
     "types-toml"
 ]
 docs = [
-    "kedro-sphinx-theme==2024.4.0",
     "ipykernel>=5.3, <7.0",
     "Jinja2<3.2.0",
+    "kedro-datasets",
```

Comment:
This is to resolve some warning that `kedro_datasets` can't be resolved. That said, it wasn't actually what was failing the build.

@stichbury or @astrojuanlu you may better know how the intersphinx mapping stuff works, and whether this should be added.

Reply:
Hmmm in principle intersphinx and pip-installing `kedro-datasets` is totally unrelated, `intersphinx` reads the inventory from the URL. What failure were you observing?

Reply:
If you see https://readthedocs.org/projects/kedro/builds/25158661/, you'll see "No module named `kedro_datasets`" in the failing block. However, this isn't what actually fails the build.

In https://readthedocs.org/projects/kedro/builds/25158721/ (after the install), you don't see the message.

Reply:
Indeed, I see it now in the 0.19.7 docs build

```
Failed to create list of (regex, reStructuredText link replacement) for class names and method names in docstrings. Error is: 
No module named 'kedro_datasets'
```